wallet/walletrpc.c

```diff
@@ -474,8 +474,8 @@ static void json_listaddrs(struct command *cmd,
 	}
 	json_array_end(response);
 	json_object_end(response);
-	command_success(cmd, response);
 	tal_free(tmpctx);
```

Comment:
Can also remove this `tal_free` now: `command_*` resolution functions destroy the `cmd`, and since `tmpctx` has `cmd` as parent, `tmpctx` is automatically freed by the `command_*` resolution functions.

Reply:
Thanks for explanation, in the [last commit](https://github.com/ElementsProject/lightning/pull/1001/commits/252d109865ff0d15568bec7cfaff15e0c2525d37) I move `tal_free` up to `command_success`. Now I can remove it.